src/workers/algorithm/worker.ts

```diff
@@ -0,0 +1,13 @@
+/* eslint-disable no-restricted-globals */
+/* Necessary because 'self' should be accepted as a global in webworkers. */
+
+import 'regenerator-runtime'
+import { run } from '../../algorithms/run'
+
+self.addEventListener('message', (event: MessageEvent) => {
+  const args = event.data
+
+  run(...args)
+    .then((result) => self.postMessage({ result }))
+    .catch((error) => self.postMessage({ error: `Algorithm error:${error.name}:${error.message}` }))
+})
```

Comment:
Here is the actual webworker code. It's a very thin wrapper around the run function.

Reply:
@ivan-aksamentov At this time, Firefox is unable to transfer Error objects via the structured clone algorithm. So this needs to be a string for now. I'll add a comment to that effect.

https://bugzilla.mozilla.org/show_bug.cgi?id=1556604